crates/server/src/authority/authority.rs

```diff
@@ -285,6 +285,15 @@ impl<T, E> LookupControlFlow<T, E> {
     pub fn is_break(&self) -> bool {
         matches!(self, Self::Break(_))
     }
+
+    /// Maps inner Ok(T) and Err(E) to Some(Result<T,E>) and Skip to None
+    pub fn map_result(self) -> Option<Result<T, E>> {
```

Comment:
Nit: I prefer to inline code additions into the commit that uses them, makes it easier to review whether it's the right fit for its intended use case.